hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -898,8 +897,13 @@ public synchronized List<String> getSSTDiffList(
       filterRelevantSstFilesFullPath(fwdDAGDifferentFiles,
           src.getTablePrefixes());
     }
-
-    return new ArrayList<>(fwdDAGDifferentFiles);
+    // Check if the DAG traversal was able to reach all the destination SST files.
+    for (String destSnapFile : destSnapFiles) {
+      if (!fwdDAGSameFiles.contains(destSnapFile) || !fwdDAGDifferentFiles.contains(destSnapFile)) {
+        return Optional.empty();
+      }
+    }
```

Comment:
Move this before if at line 896. Otherwise, it can return some delta files which can later be deleted by SstFilteringSerivce.

Reply:
Makes sense